nmpolicy/internal/capture/capture_entry.go

```diff
@@ -16,13 +16,56 @@
 
 package capture
 
-import "github.com/nmstate/nmpolicy/nmpolicy/types"
+import (
+	"fmt"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/lexer"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/parser"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/resolver"
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
 
 type CaptureEntry struct {
-	CaptureCache map[string]types.CaptureState
+	capturedStates map[string]map[string]interface{}
+	lexer          Lexer
+	parser         Parser
+	resolver       Resolver
+}
+
+func NewCaptureEntryWithLexerParserResolver(capturedStates map[string]types.CaptureState,
+	l Lexer, p Parser, r Resolver) (CaptureEntry, error) {
+	unmarshaledCapturedStates, err := unmarshalCapturedStates(capturedStates)
+	if err != nil {
+		return CaptureEntry{}, err
+	}
+	return CaptureEntry{
+		capturedStates: unmarshaledCapturedStates,
+		lexer:          l,
+		parser:         p,
+		resolver:       r,
+	}, nil
+}
+
+func NewCaptureEntry(capturedStates map[string]types.CaptureState) (CaptureEntry, error) {
```

Comment:
Since our future plan to unmarshal at the beginning of `operation.GenerateState` and marshal at the end. I would change the signature to -
`NewCaptureEntry(capturedStates map[string]map[string]interface{}) (CaptureEntry, error)`
and do the unmarshaling in `operations.GenerateState`.

EDIT: it can be done in  the follow-up that will remove the marshaled usage from all the intermediated phases.

Reply:
I am planning on doing on the follow up, it will be the next thing.